Scc mgmt org sha

vijaykanthm opened the pull request:
<!-- Put a description of what this PR is for here, along with any references to issues that this resolves or contributes to -->
Related to http://b/330538229.
Issue No: https://b.corp.google.com/issues/339718349

Requirement Doc: https://docs.google.com/document/d/1b97vduAO9BPiTTFa-muB9c9ACuDFlAOZW745PrkohXM/edit?tab=t.0#heading=h.cujlsysi3w59

Adding new template to move from SCC API v1 → SCC Management API for post-V2-GA release customers and to support
https://cloud.google.com/security-command-center/docs/reference/security-center-management/rest/v1/organizations.locations.eventThreatDetectionCustomModules

<!--
Please self-review your PR against the review checklist before creating it: https://googlecloudplatform.github.io/magic-modules/contribute/review-pr/



Completing the checklist will help speed up the review process, and we appreciate you spending time on them before sending
your code to be reviewed.

If your PR is still work in progress, please create it in draft mode
-->

<!-- AUTOCHANGELOG for Downstream PRs.

Please select one of the following "release-note:" headings:
    - release-note:enhancement
    - release-note:bug
    - release-note:note
    - release-note:new-resource
    - release-note:new-datasource
    - release-note:deprecation
    - release-note:breaking-change
    - release-note:none

Unless you choose release-note:none, please add a release note.

See https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/ for writing good release notes.

You can add more release note blocks if you want more than one CHANGELOG
entry for this PR.
-->
**Release Note Template for Downstream PRs (will be copied)**

```release-note:REPLACEME
google_scc_management_organization_security_health_analytics_custom_module
```

slevenick reviewed (changes requested):
Linting and several other steps are failing

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-10815-old..auto-pr-10815) ( 13 files changed, 2319 insertions(+), 648 deletions(-))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-10815-old..auto-pr-10815) ( 11 files changed, 2415 insertions(+), 754 deletions(-))
`terraform-google-conversion`: [Diff](https://github.com/modular-magician/terraform-google-conversion/compare/auto-pr-10815-old..auto-pr-10815) ( 1 file changed, 338 insertions(+))


## Errors

`google` provider:
- The diff processor failed to build. This is usually due to the downstream provider failing to compile.

`google-beta` provider:
- The diff processor failed to build. This is usually due to the downstream provider failing to compile.

slevenick reviewed (changes requested):
Formatting on the test is causing failures, please run make fmt to correct it

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-10815-old..auto-pr-10815) ( 13 files changed, 2319 insertions(+), 648 deletions(-))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-10815-old..auto-pr-10815) ( 11 files changed, 2415 insertions(+), 754 deletions(-))
`terraform-google-conversion`: [Diff](https://github.com/modular-magician/terraform-google-conversion/compare/auto-pr-10815-old..auto-pr-10815) ( 1 file changed, 338 insertions(+))


## Errors

`google` provider:
- The diff processor failed to build. This is usually due to the downstream provider failing to compile.

`google-beta` provider:
- The diff processor failed to build. This is usually due to the downstream provider failing to compile.

slevenick commented:
/gcbrun

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-10815-old..auto-pr-10815) ( 13 files changed, 2319 insertions(+), 648 deletions(-))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-10815-old..auto-pr-10815) ( 11 files changed, 2415 insertions(+), 754 deletions(-))
`terraform-google-conversion`: [Diff](https://github.com/modular-magician/terraform-google-conversion/compare/auto-pr-10815-old..auto-pr-10815) ( 1 file changed, 338 insertions(+))


## Errors

`google` provider:
- The diff processor failed to build. This is usually due to the downstream provider failing to compile.

`google-beta` provider:
- The diff processor failed to build. This is usually due to the downstream provider failing to compile.

slevenick reviewed (changes requested):
Looks like the package names overlap:
found packages securitycentermanagement (resource_scc_management_organization_security_health_analytics_custom_module.go) and securitycenter (resource_scc_management_organization_security_health_analytics_custom_module_test.go)

Is there a reason these are part of different products? Or can they be renamed?

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-10815-old..auto-pr-10815) ( 5 files changed, 2278 insertions(+), 651 deletions(-))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-10815-old..auto-pr-10815) ( 5 files changed, 2385 insertions(+), 758 deletions(-))
`terraform-google-conversion`: [Diff](https://github.com/modular-magician/terraform-google-conversion/compare/auto-pr-10815-old..auto-pr-10815) ( 1 file changed, 338 insertions(+))

modular-magician commented:
#### Tests analytics
Total tests: `2`
Passed tests: `1`
Skipped tests: `0`
Affected tests: `1`

<details><summary>Click here to see the affected service packages</summary><blockquote><ul><li>securitycentermanagement</li></ul></blockquote></details>


#### Action taken
<details> <summary>Found 1 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests</summary><blockquote>TestAccSecurityCenterManagementOrganizationSecurityHealthAnalyticsCustomModule </blockquote></details>

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/contributing/#general-contributing-steps)

modular-magician commented:
$\textcolor{red}{\textsf{Tests failed during RECORDING mode:}}$
`TestAccSecurityCenterManagementOrganizationSecurityHealthAnalyticsCustomModule`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-10815/artifacts/01da8df5-c515-4060-99c2-76423169e41e/build-log/recording_build/TestAccSecurityCenterManagementOrganizationSecurityHealthAnalyticsCustomModule_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-10815/artifacts/01da8df5-c515-4060-99c2-76423169e41e/recording/TestAccSecurityCenterManagementOrganizationSecurityHealthAnalyticsCustomModule.log)]


$\textcolor{red}{\textsf{Please fix these to complete your PR.}}$
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-10815/artifacts/01da8df5-c515-4060-99c2-76423169e41e/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-10815/artifacts/01da8df5-c515-4060-99c2-76423169e41e/recording) for each test

vijaykanthm commented:
Hi @slevenick, can you please have a look at the PR, I have addressed the comments and made some changes, can you approve the step to trigger the next steps in the CI. Thanks.

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-10815-old..auto-pr-10815) ( 5 files changed, 2278 insertions(+), 651 deletions(-))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-10815-old..auto-pr-10815) ( 5 files changed, 2385 insertions(+), 758 deletions(-))
`terraform-google-conversion`: [Diff](https://github.com/modular-magician/terraform-google-conversion/compare/auto-pr-10815-old..auto-pr-10815) ( 1 file changed, 338 insertions(+))

modular-magician commented:
#### Tests analytics
Total tests: `2`
Passed tests: `1`
Skipped tests: `0`
Affected tests: `1`

<details><summary>Click here to see the affected service packages</summary><blockquote><ul><li>securitycentermanagement</li></ul></blockquote></details>


#### Action taken
<details> <summary>Found 1 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests</summary><blockquote>TestAccSecurityCenterManagementOrganizationSecurityHealthAnalyticsCustomModule </blockquote></details>

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/contributing/#general-contributing-steps)

slevenick commented on mmv1/products/securitycentermanagement/OrganizationSecurityHealthAnalyticsCustomModule.yaml:
```diff
@@ -0,0 +1,228 @@
+    output: true
+    custom_flatten: templates/terraform/custom_flatten/name_from_self_link.erb
+    description: |
+      The resource name of the custom module. Its format is "organizations/{organization}/locations/{location}/securityHealthAnalyticsCustomModules/{securityHealthAnalyticsCustomModule}".
```
I'm curious why this is not user-settable. Typically resources are required to have a user settable ID field

vijaykanthm replied:
That's what the documentation says
https://cloud.google.com/security-command-center/docs/reference/security-center-management/rest/v1/organizations.locations.securityHealthAnalyticsCustomModules

It says the same in the v1 version (prior to management) documentation:
https://cloud.google.com/security-command-center/docs/reference/rest/v1/organizations.securityHealthAnalyticsSettings.customModules

So may be the docs need to be updated.

slevenick commented on mmv1/products/securitycentermanagement/OrganizationSecurityHealthAnalyticsCustomModule.yaml:
```diff
@@ -0,0 +1,228 @@
+      characters or underscores only.
+  - !ruby/object:Api::Type::Enum
+    name: 'enablementState'
+    required: false
```
required: false is the default value, so these can be removed

vijaykanthm replied:
Addressed.

slevenick commented on mmv1/third_party/terraform/services/securitycentermanagement/resource_scc_management_organization_security_health_analytics_custom_module_test.go:
```diff
@@ -0,0 +1,205 @@
+	transport_tpg "github.com/hashicorp/terraform-provider-google/google/transport"
+)
+
+// Custom Module tests cannot be run in parallel without running into 409 Conflict reponses.
```
Why is this the case? Doesn't creating a custom module create an independent resource to act on?

vijaykanthm replied:
Yes, while each custom module is indeed an independent resource, I think creating, updating, or deleting these modules in parallel may often lead to conflict responses. This is likely due to internal rate limiting, resource contention, or transactional conflicts imposed by the API's backend infrastructure. 

BTW, this comment was present in the v1 version test case, maybe they put it as disclaimer, so I put the same for this one as well.

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-10815-old..auto-pr-10815) ( 5 files changed, 2281 insertions(+), 654 deletions(-))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-10815-old..auto-pr-10815) ( 5 files changed, 2389 insertions(+), 762 deletions(-))
`terraform-google-conversion`: [Diff](https://github.com/modular-magician/terraform-google-conversion/compare/auto-pr-10815-old..auto-pr-10815) ( 1 file changed, 338 insertions(+))
